### DISCUSSION TITLE
change device type

### DISCUSSION THREAD

**Discussion by mrmoghadasi** [netbox-community/netbox#14576] [Help Wanted!] [answered]
I am using netbox version 3.3.0.
I want to change the device type of some servers.
Currently, each of these servers has interfaces connected to IPs.
Will changing the device type of these servers cause any issues with the data or IPs connected to the interfaces of these servers?

**Comment by candlerb** [comment, marked as answer]
No. When you first create a device, all its interfaces (and other components) are *copied* from the device type. Subsequent changes to the device type, or altering the device itself to point to a different device type, do not affect existing device instances.

If there are any changes to the device type that you do want to pick up - such as more or fewer interfaces or different interface names - you would have to apply these changes manually to each device, by adding/removing/renaming interfaces.

There are a couple of tools which can help here:
* https://github.com/netbox-community/customizations/blob/master/reports/dcim-reports/missing_device_type_components.py - report which identifies devices which are missing one or more interfaces in the device type
* https://github.com/netbox-community/customizations/blob/master/scripts/add_device_type_components.py - script which creates the missing interfaces (i.e. which are present in the device type but not in the device)

**Reply by mrmoghadasi** [reply]
tankyou @candlerb

**Reply by jeremystretch** [reply]
This is also explained [in the wiki](https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-arent-device-type-component-modifications-retroactive).
